conjure-java-core/src/integrationInput/java/com/palantir/product/StrictFourFields.java

```diff
@@ -0,0 +1,318 @@
+                && this.memoizedHashCode != other.memoizedHashCode) {
+            return false;
+        }
+        return this.myList.equals(other.myList)
+                && this.bearerTokenValue.equals(other.bearerTokenValue)
+                && this.optionalItem.equals(other.optionalItem)
+                && this.mappedRids.equals(other.mappedRids);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = memoizedHashCode;
+        if (result == 0) {
+            int hash = 1;
+            hash = 31 * hash + this.myList.hashCode();
+            hash = 31 * hash + this.bearerTokenValue.hashCode();
+            hash = 31 * hash + this.optionalItem.hashCode();
+            hash = 31 * hash + this.mappedRids.hashCode();
+            result = hash;
+            memoizedHashCode = result;
+        }
+        return result;
+    }
+
+    @Override
+    @DoNotLog
+    public String toString() {
+        return "StrictFourFields{myList: " + myList + ", bearerTokenValue: " + bearerTokenValue + ", optionalItem: "
+                + optionalItem + ", mappedRids: " + mappedRids + '}';
+    }
+
+    private static void validateFields(
+            List<String> myList,
+            BearerToken bearerTokenValue,
+            Optional<String> optionalItem,
+            Map<ResourceIdentifier, String> mappedRids) {
+        List<String> missingFields = null;
+        missingFields = addFieldIfMissing(missingFields, myList, "myList");
+        missingFields = addFieldIfMissing(missingFields, bearerTokenValue, "bearerTokenValue");
+        missingFields = addFieldIfMissing(missingFields, optionalItem, "optionalItem");
+        missingFields = addFieldIfMissing(missingFields, mappedRids, "mappedRids");
+        if (missingFields != null) {
+            throw new SafeIllegalArgumentException(
+                    "Some required fields have not been set", SafeArg.of("missingFields", missingFields));
+        }
+    }
+
+    private static List<String> addFieldIfMissing(List<String> prev, Object fieldValue, String fieldName) {
+        List<String> missingFields = prev;
+        if (fieldValue == null) {
+            if (missingFields == null) {
+                missingFields = new ArrayList<>(4);
+            }
+            missingFields.add(fieldName);
+        }
+        return missingFields;
+    }
+
+    public static MyListStageBuilder builder() {
+        return new DefaultBuilder();
+    }
+
+    public interface MyListStageBuilder {
+        BearerTokenValueStageBuilder myList(@Nonnull Iterable<String> myList);
+
+        Builder from(StrictFourFields other);
+    }
+
+    public interface BearerTokenValueStageBuilder {
+        OptionalItemStageBuilder bearerTokenValue(@Nonnull BearerToken bearerTokenValue);
+    }
+
+    public interface OptionalItemStageBuilder {
+        MappedRidsStageBuilder optionalItem(@Nonnull Optional<String> optionalItem);
+
+        MappedRidsStageBuilder optionalItem(@Nonnull String optionalItem);
+    }
```

Comment:
An option field's value setter is in the field builder's interface instead of in the final stage.